scripts/gen_kobject_list.py

```diff
@@ -189,33 +207,54 @@ def write_gperf_table(fp, eh, objs, static_begin, static_end):
     for i in range(0, thread_max_bytes):
         thread_idx_map[i] = 0xFF
 
+    priv_stacks = None
+    num_stacks = eh.get_thread_stack_counter()
+    stack = 0
+    object_str = ""
+    
+    if num_stacks != 0:
+        priv_stacks = "static struct _k_thread_stack_element stack_data[%d] = {\n" % num_stacks
+
     for obj_addr, ko in objs.items():
         obj_type = ko.type_name
         # pre-initialized objects fall within this memory range, they are
         # either completely initialized at build time, or done automatically
         # at boot during some PRE_KERNEL_* phase
         initialized = static_begin <= obj_addr < static_end
         is_driver = obj_type.startswith("K_OBJ_DRIVER_")
+        is_stack = obj_type == "K_OBJ__THREAD_STACK_ELEMENT"
+
+        #create privilege stack for each K_OBJ__THREAD_STACK_ELEMENT
+        if is_stack and priv_stacks:
+            if stack != 0:
+                priv_stacks += ", K_THREAD_STACK_DEFINE(stack_data[%d], %d)" % (stack, ko.data)
```

Comment:
You've already declared the privilege elevation stack buffer at line 169, this isn't what you need to do here.

Let's say there is a stack object at address 0x2001000 of size 8192 inside the kernel. Below is an example on how this could be wired up. Your code doesn't have to be exactly like this, but this should be the gist of it:

First, you need to generate the privilege stack:

```
static u8_t __used __aligned(Z_PRIVILEGE_STACK_ALIGN) priv_stack_20010000[CONFIG_PRIVILEGED_STACK_SIZE];
```

You need to generate a metadata struct:

```
/* Put this struct definition in kernel.h next to _k_object definition */
struct z_stack_metadata {
	u8_t *privilege_stack;
	size_t stack_size;
}

static z_stack_metadata stack_metadata_20010000 = {
	.privilege_stack = &priv_stack_20010000,
	.size = 8192
};
```

And then in the gperf table, an entry:

```
    {(char *)0x20010000, {}, K_OBJ__THREAD_STACK_ELEMENT, 0, (u32_t)&stack_metadata_20010000},
```

Then, provide a replacement for z_priv_stack_find() which just looks up the kobject and fetches the priv stack pointer from its linked metadata struct.

You'll also have to find all the places which are currently looking at the data value for the stack size, since the data value is now a pointer to the metadata and not the stack size itself, I think it's just thread.c:583, possibly some tests.

Finally, generation of privilege stacks needs to be optional, controlled by a flag. Currently X86 and ARC do not generate separate privilege stacks, they just roll it into the main stack object. We want to change this, but doesn't need to be in this PR. If privilege stack generation isn't enabled, continue to generate the metadata struct, but leave the .privilege_stack member NULL.